packages/gatsby/src/commands/build-html.ts

```diff
@@ -213,7 +213,8 @@ const renderHTMLQueue = async (
     [`gatsby_log_level`, process.env.gatsby_log_level],
   ]
 
-  const segments = chunk(pages, 50)
+  const chunkSize = process.env.GATSBY_IS_PREVIEW ? 1 : 50
```

Comment:
so we don't abort out of building other pages in a larger chunk. 1 paged sized chunks is a bit of a speed hit but it should be fine. We could get fancy perhaps and pick a large chunk size if `pages.length > 1000` or something.

Reply:
Doesn't your change below make it so it wont abort anyway?

Reply:
there's two aborts basically — one is out of the chunk of pages being built in a worker and then the other is aborting the entire build. I reduced the chunk size as otherwise, if the worker hit an error on page 3 of 50 pages, the remaining 46 pages won't be built.

Reply:
Got it 👍 what about putting another try/catch inside this map? https://github.com/gatsbyjs/gatsby/blob/f026e8221b331489dfc233c75d0fce14f762b944/packages/gatsby/src/commands/build-html.ts#L230-L255

In the catch we could check for GATSBY_IS_PREVIEW and reporter the error and continue the map, otherwise rethrow the error. That way we wouldn't have to change the chunkSize (unless I'm misreading the code)

Reply:
oh haha right. That would be better 👍

Reply:
Let's not reduce the chunk size here. Let's catch the error while building and continue. Batch sizing to 1 will slow things down.